### PR TITLE
Adds missing license header

### PR DESCRIPTION
**Pull request by alecharp**
@Jagrutiti to validate this content.

### Description

Classes were created and released without licenses.

### Testing done

N/A

```[tasklist]
### Submitter checklist
- [ ] If an issue exists, it is well described and linked in the description
- [x] The description of this pull request is detailed and explain why this pull request is needed
- [x] The changeset is on a specific branch. Using `feature/` for new feature, or improvements ; Using `fix/` for bug fixes ; Using `docs/` for any documentation changes.
- [ ] If required, the documentation has been updated
- [x] There is automated tests to cover the code change / addition or an explanation why there is no tests in the description.
```

### CODE REVIEW

File: core/src/main/java/io/jenkins/pluginhealth/scoring/probes/JSR305Probe.java [hunk 1]
```diff
@@ -1,3 +1,27 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022-2023 Jenkins Infra
```
Comment (Jagrutiti) [inline core/src/main/java/io/jenkins/pluginhealth/scoring/probes/JSR305Probe.java:4]: Here it is 2022-2023?

File: core/src/main/java/io/jenkins/pluginhealth/scoring/probes/RenovateProbe.java [hunk 2]
```diff
@@ -1,3 +1,27 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022-2024 Jenkins Infra
```
Comment (Jagrutiti) [inline core/src/main/java/io/jenkins/pluginhealth/scoring/probes/RenovateProbe.java:4]: Here it is 2022-2024?

**Review by Jagrutiti — COMMENTED** [review]
Shouldn't there be some consistency in all the years?

**Comment by alecharp** [comment]
The years should represent when the license was being licensed up until last edition. As the classes should have been licensed from the start, I propose to retroactively license them.

**Review by Jagrutiti — APPROVED** [review]

**Pull request merged**